fix: compile for native cuda architecture

gamemaker1 opened the pull request:
- fix the `.gitignore` file so it does not ignore the top-level directories themselves, and instead ignores the executables within any directories.
- change the target architecture/gencode for `nvcc` to `native` so that it auto-detects and builds the executable for the installed CUDA version.
- expand the documentation to include commands on building with the default (`native`) architecture and for specific architectures.

gamemaker1 committed:
fix: make sure directories are not ignored

gamemaker1 committed:
fix: compile for native cuda architecture

gamemaker1 committed:
docs: explain how to build for native and local cuda archs

ekondis commented on mixbench-cuda/CMakeLists.txt:
```diff
@@ -7,7 +7,7 @@ include_directories(${CMAKE_CUDA_TOOLKIT_INCLUDE_DIRECTORIES})
 string(APPEND CMAKE_CUDA_FLAGS " -Wno-deprecated-gpu-targets")
 string(APPEND CMAKE_CUDA_FLAGS " --cudart=static")
-string(APPEND CMAKE_CUDA_FLAGS " -gencode arch=compute_52,code=[sm_52,compute_52] -gencode arch=compute_61,code=compute_61")
+string(APPEND CMAKE_CUDA_FLAGS " -arch=native")
```
In order to actually leverage the `CMAKE_CUDA_ARCHITECTURES` cmake variable, I'd suggest to remove completely this `arch` manual setting and bump the CMake requirement to version 3.18 (line 1). This will streamline setting a specific architecture since the user will have to just set this cmake var accordingly.

gamemaker1 replied:
Done in 2c790ce27b52a3b74d5a98415a110a5c0cff0b3d.

ekondis commented on .gitignore:
nice!

gamemaker1 committed:
fix: remove `arch` from `CMAKE_CUDA_FLAGS`

- this lets the user leverage the `CMAKE_CUDA_ARCHITECTURES` macro instead
- see https://github.com/ekondis/mixbench/pull/58/commits/3624f6e04d8b8e9d03ba4f6897f11270dbcbf96d#r2573136371

gamemaker1 committed:
docs: update section on building for specific architectures
